code/modules/power/lighting.dm

```diff
@@ -385,106 +174,62 @@
+		to_chat(user, "You stick \the [I] into the light socket!")
+		var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread(src)
+		s.set_up(3, 1, loc)
+		s.start()
+		if(prob(75))
+			electrocute_mob(user, get_area(src), src, rand(7, 10) * 0.1)
 
-///flicker lights on and off
-/obj/machinery/light/proc/flicker(toggle_flicker = FALSE)
-	if(!has_power())
-		lightambient.stop(src)
+/obj/machinery/light/attacked_by(obj/item/I, mob/living/user, def_zone)
+	. = ..()
+	if(QDELETED(src))
 		return
-	if(toggle_flicker)
-		if(status != LIGHT_OK)
-			addtimer(CALLBACK(src, PROC_REF(flicker), TRUE), flicker_time)
-			return
-		flickering = !flickering
-		if(flickering)
-			lightambient.start(src)
-		else
-			lightambient.stop(src)
-	if(random_flicker)
-		flicker_time = rand(flicker_time_lower_min, flicker_time_upper_max)
-	if(status != LIGHT_OK)
-		lightambient.stop(src)
-		flickering = FALSE
-		addtimer(CALLBACK(src, PROC_REF(flicker), TRUE), flicker_time)
+	if(status != LIGHT_OK && status != LIGHT_BURNED)
 		return
-	light_flicker_state = !light_flicker_state
-	if(!light_flicker_state)
-		flick("[base_icon_state]_flick_off", src)
-		//delay the power change long enough to get the flick() animation off
-		addtimer(CALLBACK(src, PROC_REF(flicker_power_state)), 0.3 SECONDS)
-	else
-		flick("[base_icon_state]_flick_on", src)
-		addtimer(CALLBACK(src, PROC_REF(flicker_power_state)), 0.3 SECONDS)
-		flicker_time = flicker_time * 2 //for effect it's best if the amount of time we spend off is more than the time we spend on
-	if(!flickering)
+	if(!prob(1 + I.force * 5))
 		return
-	addtimer(CALLBACK(src, PROC_REF(flicker)), flicker_time)
 
-///proc to toggle power on and off for light
-/obj/machinery/light/proc/flicker_power_state(turn_on = TRUE, turn_off = FALSE)
-	if(!light_flicker_state)
-		pick(playsound(loc, 'sound/effects/lightfizz.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz2.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz3.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz4.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz5.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz6.ogg', 10, TRUE))
-		update(FALSE)
-	else
-		pick(playsound(loc, 'sound/effects/lightfizz.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz2.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz3.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz4.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz5.ogg', 10, TRUE), playsound(loc, 'sound/effects/lightfizz6.ogg', 10, TRUE))
-		turn_light(null, FALSE)
+	visible_message("[user] smashed the light!", "You hit the light, and it smashes!")
+	if(light_on && (I.atom_flags & CONDUCT) && prob(12))
+		electrocute_mob(user, get_area(src), src, 0.3)
+	broken()
+
+/obj/machinery/light/screwdriver_act(mob/living/user, obj/item/I)
+	if(user.a_intent == INTENT_HARM)
+		return FALSE
+	if(status != LIGHT_EMPTY)
+		balloon_alert(user, "Remove bulb")
```

Comment:
Was confused for a bit but this is an instruction rather than an action, yeah? Feels unclear to me.

Reply:
updated